### PR TITLE
Rename two Snapshot classes to reflect what they contain

### PR DESCRIPTION
**Pull request by JHolba**
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/ensemble_evaluator/_ensemble.py [hunk 1]
```diff
@@ -123,15 +123,15 @@ def __post_init__(self) -> None:
...
+        snapshot = EnsembleSnapshot()
         snapshot._ensemble_state = ENSEMBLE_STATE_UNKNOWN
         for real in self.active_reals:
             realization = RealizationSnapshot(
-                active=True, status=REALIZATION_STATE_WAITING, forward_models={}
+                active=True, status=REALIZATION_STATE_WAITING, fm_steps={}
             )
-            for index, forward_model in enumerate(real.forward_models):
-                realization["forward_models"][str(index)] = ForwardModel(
+            for index, forward_model in enumerate(real.fm_steps):
```
Comment (jonathan-eq) [inline src/ert/ensemble_evaluator/_ensemble.py:133]:
```suggestion
            for index, forward_model_step in enumerate(real.fm_steps):
```

File: src/ert/ensemble_evaluator/evaluator.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ async def _stopped_handler(self, events: Sequence[EnsembleSucceeded]) -> None:
             return
 
         max_memory_usage = -1
-        for job in self.ensemble.snapshot.get_all_forward_models().values():
+        for job in self.ensemble.snapshot.get_all_fm_steps().values():
```
Comment (jonathan-eq) [inline src/ert/ensemble_evaluator/evaluator.py:164]:
```suggestion
        for forward_model_step in self.ensemble.snapshot.get_all_fm_steps().values():
```

File: src/ert/ensemble_evaluator/snapshot.py [hunk 3]
```diff
@@ -91,84 +91,77 @@ def convert_iso8601_to_datetime(
...
+    aggr_fm_step_status_colors: DefaultDict[RealId, Dict[FmStepId, QColor]]
+    # contains the QColor used in the GUI for each real
     real_status_colors: Dict[RealId, QColor]
     sorted_real_ids: List[RealId]
-    sorted_forward_model_ids: DefaultDict[RealId, List[FmStepId]]
+    sorted_fm_step_ids: DefaultDict[RealId, List[FmStepId]]
 
 
-class Snapshot:
+class EnsembleSnapshot:
```
Comment (jonathan-eq) [inline src/ert/ensemble_evaluator/snapshot.py:103]: Fantastic :)

File: src/ert/ensemble_evaluator/snapshot.py [hunk 4]
```diff
@@ -198,55 +191,54 @@ def status(self) -> Optional[str]:
...
         self,
-    ) -> Mapping[Tuple[RealId, FmStepId], "ForwardModel"]:
-        return self._forward_model_states.copy()
+    ) -> Mapping[Tuple[RealId, FmStepId], "FMStepSnapshot"]:
+        return self._fm_step_snapshots.copy()
 
-    def get_forward_model_status_for_all_reals(
+    def get_fm_steps_for_all_reals(
         self,
     ) -> Mapping[Tuple[str, str], str]:
```
Comment (jonathan-eq) [inline src/ert/ensemble_evaluator/snapshot.py:204]:
```suggestion
    ) -> Mapping[Tuple[RealId, FmStepId], str]:
```

File: src/ert/ensemble_evaluator/snapshot.py [hunk 5]
```diff
@@ -91,84 +91,77 @@ def convert_iso8601_to_datetime(
...
-            snapshot._ensemble_state = data["status"]
-        for real_id, realization_data in data.get("reals", {}).items():
-            snapshot.add_realization(
+    def from_nested_dict(cls, snapshot: Mapping[Any, Any]) -> "EnsembleSnapshot":
+        new_snapshot = EnsembleSnapshot()
+        if "metadata" in snapshot:
+            new_snapshot._metadata = snapshot["metadata"]
+        if "status" in snapshot:
+            new_snapshot._ensemble_state = snapshot["status"]
+        for real_id, realization_data in snapshot.get("reals", {}).items():
```
Comment (jonathan-eq) [inline src/ert/ensemble_evaluator/snapshot.py:137]:
```suggestion
    def from_nested_dict(cls, input_dict: Mapping[Any, Any]) -> "EnsembleSnapshot":
        new_snapshot = EnsembleSnapshot()
        if "metadata" in input_dict:
            new_snapshot._metadata = input_dict["metadata"]
        if "status" in snapshot:
            new_snapshot._ensemble_state = input_dict["status"]
        for real_id, realization_data in input_dict.get("reals", {}).items():
```

File: src/ert/ensemble_evaluator/snapshot.py [hunk 6]
```diff
@@ -198,55 +191,54 @@ def status(self) -> Optional[str]:
...
-            for idx, forward_model_state in self._forward_model_states.items()
-            if "status" in forward_model_state
-            and forward_model_state["status"] is not None
+            idx: fm_step_snapshot["status"]
+            for idx, fm_step_snapshot in self._fm_step_snapshots.items()
+            if "status" in fm_step_snapshot and fm_step_snapshot["status"] is not None
         }
 
     @property
     def reals(self) -> Mapping[str, "RealizationSnapshot"]:
```
Comment (jonathan-eq) [inline src/ert/ensemble_evaluator/snapshot.py:212]:
```suggestion
    def reals(self) -> Mapping[RealId, "RealizationSnapshot"]:
```

File: src/ert/gui/model/job_list.py [hunk 7]
```diff
@@ -13,10 +13,10 @@
 
 from ert.ensemble_evaluator import identifiers as ids
 from ert.gui.model.snapshot import (
-    JOB_COLUMN_SIZE,
-    JOB_COLUMNS,
+    FM_STEP_COLUMN_SIZE,
```
[the remark this answers is not in the document]
Comment (JHolba) [reply src/ert/gui/model/job_list.py:16]: good point. renamed it.

File: src/ert/ensemble_evaluator/snapshot.py (hunk 4 above)
Comment (JHolba) [reply src/ert/ensemble_evaluator/snapshot.py]: fixed

File: src/ert/ensemble_evaluator/snapshot.py (hunk 5 above)
Comment (JHolba) [reply src/ert/ensemble_evaluator/snapshot.py]: changed it from snapshot to source

File: src/ert/ensemble_evaluator/snapshot.py (hunk 6 above)
Comment (JHolba) [reply src/ert/ensemble_evaluator/snapshot.py]: fixed

**Comment by xjules** [comment]
Not sure if required, but this has file/test has not been touched `tests/integration_tests/test_cli.py::test_that_log_is_cleaned_up_from_repeated_forward_models`

**Review by jonathan-eq — CHANGES_REQUESTED** [review]
Tests broke

**Review by jonathan-eq — APPROVED** [review]
Very good!

**Pull request merged**